src/urllib3/util/url.py

```diff
@@ -164,46 +177,27 @@ def url(self):
 
         return url
 
-    def __str__(self):
+    def __str__(self) -> str:
         return self.url
 
 
-def split_first(s, delims):
-    """
-    .. deprecated:: 1.25
+@typing.overload
+def _encode_invalid_chars(
+    component: str, allowed_chars: typing.Container[str]
+) -> str:  # Abstract
+    ...
 
-    Given a string and an iterable of delimiters, split on the first found
-    delimiter. Return two split parts and the matched delimiter.
 
-    If not found, then the first part is the full input string.
-
-    Example::
-
-        >>> split_first('foo/bar?baz', '?/=')
-        ('foo', 'bar?baz', '/')
-        >>> split_first('foo/bar?baz', '123')
-        ('foo/bar?baz', '', None)
-
-    Scales linearly with number of delims. Not ideal for large number of delims.
-    """
-    min_idx = None
-    min_delim = None
-    for d in delims:
-        idx = s.find(d)
-        if idx < 0:
-            continue
+@typing.overload
+def _encode_invalid_chars(
+    component: None, allowed_chars: typing.Container[str]
+) -> None:  # Abstract
+    ...
```

Comment:
TIL about typing.overload, thanks!

(It would have been cleaner to never call `_encode_invalid_chars` with a `None` component, but maybe that would have resulted in a lot of boilerplate in callers, and anyway let's not change the code when adding types.)